src/server.ts

```diff
@@ -127,9 +51,7 @@ export class Server extends events.EventEmitter {
 	}
 
 	public readonly options: ServerOptions;
-	private __running: boolean;
-	private __instance: ChildProcess;
-	private readonly __argMapping = {
+	protected readonly __argMapping = {
```

Comment:
should we make argmapping abstracted/required for instantiating super class?

Reply:
It already is, it's defined as an abstract property in the `AbstractService` class, so if not defined the class doesn't even compile. Is that what you mean?

Reply:
ah, that seems a bit weird as a readonly property.  I'd probably add it as an argument of the AbstractService constructor or possibly as an abstract get function.

Reply:
Fixed.